packages/webpack-cli/lib/utils/cli-flags.js

```diff
@@ -6,6 +8,16 @@ const ADVANCED_GROUP = 'advanced';
 const DISPLAY_GROUP = 'stats';
 const ZERO_CONFIG_GROUP = 'zero-config';
 
+const flagsFromCore = version.startsWith(5)
+    ? Object.entries(require('webpack').cli.getArguments()).map(([flag, meta]) => {
+          return {
+              ...meta,
+              name: flag,
+              group: 'core',
+          };
+      })
+    : [];
```

Comment:
Can you provide list of all options here?

Reply:
100+ options are exported from the core as listed here - 

https://github.com/webpack/webpack/blob/master/test/__snapshots__/Cli.test.js.snap

Reply:
Yep not possible to list options, some 230 flags are present in core, link is better

Reply:
Ideally we should test all of them, so I prefer to write one test like:
```
const optionsAndValues = [...];


for (const case of (optionsAndValues)) {
  it(`${case}`, () => {
   // Base test with `WebpackCLITestPlugin` and check we set options correctly
  })
}

```

Maybe we should put it in `integration` tests

Reply:
After this, we will catch all broken options on our and webpack side, maybe some of them should be excluded, because webpack in `beate`

Reply:
We can write one test with 3 test cases one each for type `string` `boolean` and `number`

Reply:
Yep :+1:

Reply:
@evilebottnawi there are a total of 290 flags and it is very difficult to write a base test for all of them because we have to write different tests assertions for most flags as some flags depends on other flags, for some property names are not same as the flag name, for example, if we want to write a base test for all boolean flags -

- for `--bail` option name is `bail` as well and it should set `bail: true`.
- for `--dependencies-reset` option name is `dependencies` and it should set `dependencies: []`  but for `output-enabled-library-types-reset` property name is `enabledLibraryTypes`
- for `--node-global` option is global => `global: true`.

Hence instead we can group flags on the basis of primary config option, for example, there are 56 flags for `resolve` related config we can tests these together. similarly, around 50 flags related to `output`, 46 flags related to `optimization`.